neuraxle/base.py

```diff
@@ -969,6 +999,33 @@ def split(self, step_type: type) -> List['TruncableSteps']:
     def ends_with(self, step_type: type):
         return isinstance(self[-1], step_type)
 
+    def set_train(self, is_train: bool=True) -> 'BaseStep':
+        """
+        Set pipeline step mode to train or test.
+
+        In the pipeline steps functions, you can add a simple if statement to direct to the right implementation:
+        `
+            def transform(self, data_inputs):
+                if self.is_train:
+                    self.transform_train_(data_inputs)
+                else:
+                    self.transform_test_(data_inputs)
+
+            def fit_transform(self, data_inputs, expected_outputs):
+                if self.is_train:
+                    self.fit_transform_train_(data_inputs, expected_outputs)
+                else:
+                    self.fit_transform_test_(data_inputs, expected_outputs)
+        `
+
+        :param is_train: bool
+        :return: self
+        """
+        self.is_train = is_train
+        for _, step in self.items():
```

Comment:
I won't request this for the current PR, but it'd be cool to be able to do `self.values()` and `self.keys` as well as `self.items()`. The values function would have been useful here for instance.

Reply:
created issue #107